modules/apps/headless/headless-site/headless-site-impl/src/main/java/com/liferay/headless/site/internal/resource/v1_0/SiteResourceImpl.java

```diff
@@ -463,15 +464,34 @@ private int _getMembershipRestriction(Integer membershipRestriction) {
 		return membershipRestriction;
 	}
 
-	private long _getParentGroupId(String parentSiteKey)
-		throws PortalException {
-
-		if (Validator.isNull(parentSiteKey)) {
+	private long _getParentGroupId(Group group, String parentSiteERC) {
+		if (Validator.isNull(parentSiteERC)) {
 			return GroupConstants.DEFAULT_PARENT_GROUP_ID;
 		}
 
-		Group parentGroup = _groupLocalService.getGroup(
-			contextCompany.getCompanyId(), parentSiteKey);
+		Group parentGroup =
+			_groupLocalService.fetchGroupByExternalReferenceCode(
+				parentSiteERC, contextCompany.getCompanyId());
+
+		if (LazyReferencingThreadLocal.isEnabled()) {
```

Comment:
If possible this check should be inverted in order to return early and buy space for the rest of the method (see https://github.com/liferay/liferay-portal/commit/332206d5ab8a518232685e6771ceec60f4720305).